controllers/toolchainconfig/sync_test.go

```diff
@@ -0,0 +1,205 @@
+package toolchainconfig
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	. "github.com/codeready-toolchain/host-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func TestSyncMemberConfigs(t *testing.T) {
+	defaultMemberConfig := config.NewMemberOperatorConfig(config.MemberStatus().RefreshPeriod("5s"))
+	specificMemberConfig := config.NewMemberOperatorConfig(config.MemberStatus().RefreshPeriod("10s"))
+
+	t.Run("sync success", func(t *testing.T) {
+
+		t.Run("no member clusters available - skip sync", func(t *testing.T) {
+			// given
+			toolchainConfig := config.NewToolchainConfig(config.AutomaticApproval().Enabled().MaxUsersNumber(123, config.PerMemberCluster("member1", 321)), config.Members().Default(defaultMemberConfig.Spec), config.Members().SpecificPerMemberCluster("member2", specificMemberConfig.Spec))
```

Comment:
such a line is super long and hard to read - could you please break it for example to:
```suggestion
			toolchainConfig := config.NewToolchainConfig(
				config.AutomaticApproval().Enabled().MaxUsersNumber(123, 
					config.PerMemberCluster("member1", 321)), 
					config.Members().Default(defaultMemberConfig.Spec), 
					config.Members().SpecificPerMemberCluster("member2", specificMemberConfig.Spec))
```

Reply:
Just out of curiosity - why do we need to specify the max number of users per member cluster `config.PerMemberCluster("member1", 321)`? And why do we need to specify the AutomaticApproval configuration at all? I guess that this is not needed for the purpose of these tests.

Reply:
Sure, I can do that to make it easier to read. No, it's not needed but I figured the majority of the time the toolchainconfig resource would have those values set so I included it but I can remove it since it might be confusing. Done in 83a653b2b5db2f88a73a770838fac9536c6a3ffd